### PR TITLE
feat: add aws support for github action

### PR DESCRIPTION
**Pull request by trathailoi**
### **PR Type**
Enhancement

___

### **Description**
- Added support for AWS credentials in the GitHub Action runner by setting environment variables `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_REGION_NAME`.
- Included assertions to ensure that AWS credentials are complete before setting the environment variables.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>github_action_runner.py</strong><dd><code>Add AWS credentials support in GitHub Action runner</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/servers/github_action_runner.py

<li>Added AWS credentials support by setting environment variables.<br> <li> Included assertions to ensure AWS credentials are complete.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1060/files#diff-15f9c11da031449998128bdbba057d768853e836eedf44563be71b3847be80e0">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: pr_agent/servers/github_action_runner.py [hunk 1]
```diff
@@ -58,6 +58,13 @@ async def run_action():
         print("OPENAI_KEY not set")
     if OPENAI_ORG:
         get_settings().set("OPENAI.ORG", OPENAI_ORG)
+
+    if get_settings().get("aws.AWS_ACCESS_KEY_ID"):
+        assert get_settings().aws.AWS_SECRET_ACCESS_KEY and get_settings().aws.AWS_REGION_NAME, "AWS credentials are incomplete"
+        os.environ["AWS_ACCESS_KEY_ID"] = get_settings().aws.AWS_ACCESS_KEY_ID
+        os.environ["AWS_SECRET_ACCESS_KEY"] = get_settings().aws.AWS_SECRET_ACCESS_KEY
+        os.environ["AWS_REGION_NAME"] = get_settings().aws.AWS_REGION_NAME
```
Comment (KennyDizi) [inline pr_agent/servers/github_action_runner.py:66]: Why do we need to inject `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_REGION_NAME` to environment variables? @trathailoi

Comment (mrT23) [reply pr_agent/servers/github_action_runner.py:66]: @trathailoi 

@KennyDizi is correct. with github action, just give the values as direct env variables.

Comment (trathailoi) [reply pr_agent/servers/github_action_runner.py:66]: I've tried many times, many ways, but didn't work.

Comment (mrT23) [reply pr_agent/servers/github_action_runner.py:66]: aws is [supported](https://github.com/Codium-ai/pr-agent/issues/1050#issuecomment-2244241459) in github action, so this PR will not be merged.

you can try pr-agent pro if you prefer a hosted version of PR-Agent, provided by CodiumAI.

all the best

Comment (trathailoi) [reply pr_agent/servers/github_action_runner.py:66]: it always tries to look for OPENAI_KEY, I don't set it though.
Screenshot 2024-07-26 at 22 37 12
Screenshot 2024-07-26 at 22 38 26

Comment (mrT23) [reply pr_agent/servers/github_action_runner.py:66]: where does the error message say anything about openAI ?

Comment (trathailoi) [reply pr_agent/servers/github_action_runner.py:66]: the part saying `read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.'`. I meant to use aws bedrock, but the error was telling me to check out the OpenAI docs.

Comment (trathailoi) [reply pr_agent/servers/github_action_runner.py:66]: I'm so sorry, I meant to fork and create the PR on that fork to test the approach myself, but I accidentally create this one. I'm going to closed this PR.

Comment (mrT23) [reply pr_agent/servers/github_action_runner.py:66]: i cannot debug you personal deployment.
somewhere you are giving an openai key.
maybe in your secrets, or in a configuration file.

debug from CLI.

In any case, this PR is plain wrong

Comment (trathailoi) [reply pr_agent/servers/github_action_runner.py:66]: you are right, I was setting openai key in my secrets. I removed it, and the 429 error code was gone.
but now it is 401 instead 😂
Screenshot 2024-07-26 at 23 36 43

Comment (trathailoi) [reply pr_agent/servers/github_action_runner.py:66]: I can confirm that the openai key was not set. But looks like the env variables that I set was not working.
Screenshot 2024-07-26 at 23 30 13

here is my setting
Screenshot 2024-07-26 at 23 29 29
Screenshot 2024-07-26 at 23 29 03

I don't put any configuration file in the repo btw.

Comment (trathailoi) [reply pr_agent/servers/github_action_runner.py:66]: could you please give me some advice on any next things that I should do to investigate the issue ? 😢

**Pull request closed without merging**